src/Compilers/CSharp/Test/Semantic/Semantics/StructConstructorTests.cs

```diff
@@ -2696,6 +2755,482 @@ .maxstack  2
+
+            verifier.VerifyIL("S..ctor", @"
+{
+  // Code size       21 (0x15)
+  .maxstack  2
+  // sequence point: {
+  IL_0000:  nop
+  // sequence point: <hidden>
+  IL_0001:  ldarg.0
+  IL_0002:  ldc.i4.0
+  IL_0003:  stfld      ""int S.X""
+  // sequence point: X.ToString();
+  IL_0008:  ldarg.0
+  IL_0009:  ldflda     ""int S.X""
+  IL_000e:  call       ""string int.ToString()""
+  IL_0013:  pop
+  // sequence point: }
+  IL_0014:  ret
+}
+",
+                sequencePoints: "S..ctor",
+                source: source);
+        }
+
+        [Fact]
+        public void ImplicitlyInitializedFields_SequencePoints_ExpressionBody()
+        {
+            var source = @"
+public struct S
+{
+    public int X;
+
+    public S()
+        => X.ToString();
+}";
+            var verifier = CompileAndVerify(source, options: TestOptions.DebugDll.WithSpecificDiagnosticOptions(ReportStructInitializationWarnings));
+            verifier.VerifyDiagnostics(
+                // (6,12): warning CS9022: Control is returned to caller before field 'S.X' is explicitly assigned, causing a preceding implicit assignment of 'default'.
+                //     public S()
+                Diagnostic(ErrorCode.WRN_UnassignedThisSupportedVersion, "S").WithArguments("S.X").WithLocation(6, 12),
+                // (7,12): warning CS9019: Field 'X' is read before being explicitly assigned, causing a preceding implicit assignment of 'default'.
+                //         => X.ToString();
+                Diagnostic(ErrorCode.WRN_UseDefViolationFieldSupportedVersion, "X").WithArguments("X").WithLocation(7, 12));
+
+            verifier.VerifyIL("S..ctor", @"
+{
+  // Code size       20 (0x14)
+  .maxstack  2
+  // sequence point: <hidden>
+  IL_0000:  ldarg.0
+  IL_0001:  ldc.i4.0
+  IL_0002:  stfld      ""int S.X""
+  // sequence point: X.ToString()
+  IL_0007:  ldarg.0
+  IL_0008:  ldflda     ""int S.X""
+  IL_000d:  call       ""string int.ToString()""
+  IL_0012:  pop
+  IL_0013:  ret
+}
+",
+                sequencePoints: "S..ctor",
+                source: source);
+        }
+
+        [Fact]
+        public void ImplicitlyInitializedFields_PragmaRestore()
+        {
+            var source = @"
+public struct S
+{
+    public int X;
+#pragma warning restore CS9022
+    public S()
+    {
+    }
+}";
+            var comp = CreateCompilation(source, options: TestOptions.DebugDll.WithSpecificDiagnosticOptions(ReportStructInitializationWarnings));
+            comp.VerifyDiagnostics(
+                // (6,12): warning CS9022: Control is returned to caller before field 'S.X' is explicitly assigned, causing a preceding implicit assignment of 'default'.
+                //     public S()
+                Diagnostic(ErrorCode.WRN_UnassignedThisSupportedVersion, "S").WithArguments("S.X").WithLocation(6, 12));
+
+            comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Consider testing `pragma warning disable 9022` as well, with and without `WithSpecificDiagnosticOptions(ReportStructInitializationWarnings)`.